### PR TITLE
added gitignore

### PR DESCRIPTION
**Pull request by anirudTT**
### Added .gitignore file

This PR adds a .gitignore file to ignore environment directories and specific `.whl` files used for `pybuda i`or other installations.

### CODE REVIEW

File: .gitignore
Comment (milank94) [inline .gitignore:1]: There is a `.gitignore` file in the `model_demos` directory already: https://github.com/tenstorrent/tt-buda-demos/blob/main/model_demos/.gitignore

I suggested making the change of pulling that one into the root directory and then adding in the extra ignores

Comment (milank94) [reply .gitignore:1]: Can you check if moving the `.gitignore` file from `model_demos` to `root` will effectively ignore the TT artifact files that get generated if running from within `model_demos`?

* cd `model_demos`
* Run any demo i.e. `python nlp_demos/bert/pytorch_bert_masked_lm.py`
* Should automatically generate `generated_modules`, `tt_build`, `*netlist.yaml`
* Check that these are being ignored

Comment (anirudTT) [reply .gitignore:1]: test

**Review by milank94 — APPROVED** [review]
LGTM

**Review by anirudTT — COMMENTED** [review]
added the changes

**Pull request merged**